src/lib/Middleware.js

```diff
@@ -1,35 +1,54 @@
 // We are not using NextJS's native middleware feature
 // because Amplify might not support it.
 //
-import { getCookie } from 'cookies-next';
 import { verifyToken } from "../lib/JWT";
+import InvalidTokenError from "../errors/InvalidTokenError";
 import MissingHeaderError from "../errors/MissingHeaderError";
 
+const BEARER_LENGTH = 7;
+
 const tokenValidator = async (req, res) => {
-  const { token } = req;
+    const { token } = req;
 
-  if (!token) throw new MissingHeaderError();
+    if (!token) throw new MissingHeaderError();
+    if (!token.startsWith("Bearer")) throw new InvalidTokenError();
 };
 
-const tokenVerifier = async (req, res) => {
-  const { token } = req;
-
-  verifyToken(token);
+const tokenExtractor = async (req, res) => {
+    req.token = req.token.substring(BEARER_LENGTH);
 };
 
-const checkToken = async (req, res) => {
-  const token = getCookie('token', { req, res });
+const tokenVerifier = async (req, res) => {
+    const { token } = req;
 
-  if (req.method === "GET" && !token) return;
+    const user = verifyToken(token);
 
-  req.token = token;
+    req.user = user;
+};
 
-  await tokenValidator(req, res);
-  await tokenVerifier(req, res);
+const checkToken = async (req, res) => {
+    const { authorization } = req.headers;
+
+    // Allows public access to all data except for user data
+    if (req.method === "GET" && !authorization) {
+        if (!req.url.includes(`/api/admin/users`)) return;
+    }
+    // Allows users to sign in/up
+    if (req.method === "POST" && !authorization) {
+        if (req.url === '/api/admin/users' || req.url === '/api/session') return;
+    }
```

Comment:
I'm not in favor of this for 2 reasons:

1.) We should not be implementing "blacklists", where we allow users to view except some items in the list. The more safer approach would be a "whitelist", wherein public access is NOT allowed by default, unless we add them into the whitelist.

2.) This is redundant if we put a ` await assertUserCan(permissions, req);` at the start of every API endpoint. 

I suggest this be removed.

Reply:
Resolved: [6522402](https://github.com/klaudsol/klaudsol-commons/pull/36/commits/65224020be18ede626af4f5f2ececab8dc03374d)